### PR TITLE
[13.0][MIG] hr_job_category

### PR DESCRIPTION
**Pull request by jarroyomorales**
Migration of `hr_job_category` and simplified code.

@etobella @Nikul-Chaudhary @i-vyshnevska @alexey-pelykh

### CODE REVIEW

File: hr_job_category/models/hr.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    @api.model
+    def create(self, vals):
+        res = super().create(vals)
+        if "job_id" in vals:
+            res._tag_employees(vals.get("job_id"))
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "job_id" in vals:
```
Comment (i-vyshnevska) [inline hr_job_category/models/hr.py]: if the employee was changed you are not removing tags from him

Comment (jarroyomorales) [reply hr_job_category/models/hr.py]: Done!

File: hr_job_category/models/hr.py [hunk 2]
```diff
@@ -23,57 +23,27 @@ class HRJob(models.Model):
...
-                ):
-                    self._remove_tags(contract.employee_id.id, data["job_id"][0])
-                self._tag_employees(contract.employee_id.id, contract.job_id.id)
+        if "job_id" in vals or ("employee_id" in vals and vals["employee_id"]):
+            self._tag_employees(vals.get("job_id"))
         return res
+
+    def unlink(self):
+        self.mapped("employee_id").write({"category_ids": [(5,)]})
+        return super().unlink()
```
Comment (dreispt) [inline hr_job_category/models/hr.py:54]: If you're deleting the record, what is the point of resetting `category_ids`?

File: hr_job_category/tests/test_hr_job_categories.py [hunk 3]
```diff
@@ -14,7 +14,8 @@ def setUp(self):
         self.contract_model = self.env["hr.contract"]
 
         # Create a employee
-        self.employee_id = self.employee_model.create({"name": "Employee 1"})
+        self.employee_id_1 = self.employee_model.create({"name": "Employee 1"})
+        self.employee_id_2 = self.employee_model.create({"name": "Employee 1"})
```
Comment (dreispt) [inline hr_job_category/tests/test_hr_job_categories.py]:
```suggestion
        self.employee_id_2 = self.employee_model.create({"name": "Employee 2"})
```

**Review by dreispt — APPROVED** [review]
LGTM

File: hr_job_category/models/hr.py (hunk 2 above)
Comment (jarroyomorales) [reply hr_job_category/models/hr.py:54]: Okay, I will keep it

File: hr_job_category/tests/test_hr_job_categories.py (hunk 3 above)
Comment (jarroyomorales) [reply hr_job_category/tests/test_hr_job_categories.py]: Thank you!

**Review by etobella — APPROVED** [review]

**Pull request merged**